app/views/my_modules/repositories/_full_view_version.html.erb

```diff
@@ -28,9 +28,11 @@
       <% end %>
     </div>
     <div class="col-xs-4">
-      <a class="pull-right btn version-button delete-snapshot-button" href="#" data-action-path="<%= my_module_repository_snapshot_path(@my_module, repository_snapshot) %>">
-        <i class="fas fa-trash"></i>
-      </a>
+      <% if can_manage_module?(@my_module) %>
```

Comment:
This check will be executed separately for each snapshot? Can you move it out from the loop and use local variable?

Reply:
Updated. Anyway, the value of this check is loaded in the cache and it's not querying each iteration